### PR TITLE
Refactor: CommonItemActionPo

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

Share the PO among all the ItemAction components that use the CommonItemAction.

# Changes

* New PO CommonItemActionPo.
* All the others extend this one.
* Remove unnecessary `span` to access title with data-tid in all CommonItemAction implementations.
* Adapt all tests and instance creations to the new interface.
* New test for CommonItemAction.

# Tests

Only test changes.

# Todos

Covered by a changelog entry.

### CODE REVIEW

File: frontend/src/tests/page-objects/NnsAvailableMaturityItemAction.page-object.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 import type { PageObjectElement } from "$tests/types/page-object.types";
+import { CommonItemActionPo } from "./CommonItemAction.page-object";
 
-export class NnsAvailableMaturityItemActionPo extends BasePageObject {
+export class NnsAvailableMaturityItemActionPo extends CommonItemActionPo {
   private static readonly TID = "nns-available-maturity-item-action-component";
 
-  static under(element: PageObjectElement): NnsAvailableMaturityItemActionPo {
+  static under({
+    element,
```
Comment (dskloetd) [inline frontend/src/tests/page-objects/NnsAvailableMaturityItemAction.page-object.ts:8]: Why did you change the argument to an object while it's still only 1 argument?

(Same for the other POs.)

File: frontend/src/tests/lib/components/neuron-detail/NnsAvailableMaturityItemAction.spec.ts [hunk 2]
```diff
@@ -52,7 +52,7 @@ describe("NnsAvailableMaturityItemAction", () => {
     };
     const po = renderComponent(neuron);
 
-    expect(await po.getMaturity()).toBe("3.14");
+    expect(await po.getTitle()).toBe("3.14");
```
Comment (dskloetd) [inline frontend/src/tests/lib/components/neuron-detail/NnsAvailableMaturityItemAction.spec.ts:55]: I'm not so sure about having the method to get the maturity be called `getTitle()`. I might suggest adding back a `getMaturity` method which just returns `this.getTitle()` but then the benefit of the share base class becomes very limited.

I originally suggested merging the Nns and Sns page objects of the same item. I'm not so convinced about `CommonItemActionPo`.

(This comment applies to all the other `getXXX()` methods as well.)

File: frontend/src/tests/page-objects/NnsNeuronStateItemAction.page-object.ts
Comment (dskloetd) [inline frontend/src/tests/page-objects/NnsNeuronStateItemAction.page-object.ts:1]: Should `getState()` now return `this.getTitle()` or be removed?

**Review by dskloetd — COMMENTED** [review]
I'm not sure I like this. It looks like a big change with limited benefit and increased complexity. Is there a follow-up planned that will make it worth it?

File: frontend/src/tests/page-objects/NnsAvailableMaturityItemAction.page-object.ts (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/tests/page-objects/NnsAvailableMaturityItemAction.page-object.ts:8]: Because it extends from `CommonItemActionPo` which expects both testId and element.

File: frontend/src/tests/lib/components/neuron-detail/NnsAvailableMaturityItemAction.spec.ts (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/tests/lib/components/neuron-detail/NnsAvailableMaturityItemAction.spec.ts:55]: I agree with you that it doesn't add much value.

I don't agree that we should merge POs if they don't map to the component structure.

**Comment by lmuntaner** [comment]
This PR and changes don't add as much value as we thought they would. Closing it.
